source/wp-content/themes/wporg-parent-2021/sass/blocks/_spinner.scss

```diff
@@ -0,0 +1,49 @@
+.wporg-spinner {
+	--local-size: 16px;
+	--spinner-color: var(--wp--preset--color--blueberry-1, currentColor);
```

Comment:
I misunderstood this, and tried to change the color of the spinning bit by putting the span in a paragraph block and changing the text color to orange, which didn't work. If we reversed this that would work, but would default to charcoal most of the time. Maybe that's ok, in allowing use to have editor customisation.

Use case would be on a page like SotW, where we show the map of watch parties and there is usually a custom link color.

![Screenshot 2024-12-09 at 3 16 50 PM](https://github.com/user-attachments/assets/cd2cec07-4429-431e-8b30-8c5e3aa5a25f)

Reply:
> If we reversed this that would work, but would default to charcoal most of the time.

I don't know if that's possible seeing that currentColor would always be set or inherited so we would never get the blueberry coloring unless explicitly set. 

We could do something like this though:
```
--spinner-color: var(--wp--custom--wporg-spinner--color, var(--wp--preset--color--blueberry-1, currentColor));
```

We could then set `--wp--custom--wporg-spinner--color` to give it a specific color.

Reply:
> --spinner-color: var(--wp--preset--color--blueberry-1, currentColor);

Since this CSS exists in the parent theme, `--wp--preset--color--blueberry-1` will always exist. So it will never use `currentColor`.

> Maybe that's ok, in allowing us to have editor customisation.

Do we need _editor_ customization? This seems like a helper class we might use in another block, where we could override with custom properties in the containing block. That said, I would change it to populate the custom properties from theme.json, for consistency with other block settings.

```json
"settings": {
	"custom": {
		"wporg-spinner": {
			"color": {
				"background": "var(--wp--preset--color--black-opacity-15)",
				"stroke": "var(--wp--preset--color--blueberry-1)"
			}
		}
```

Which would generate `--wp--custom--wporg-spinner--color--background` & `--wp--custom--wporg-spinner--color--stroke`, so you don't need to set them in the CSS either.